MCprep_addon/spawner/mcmodel.py

```diff
@@ -694,6 +723,57 @@ def execute(self, context):
 		update_model_list(context)
 		return {'FINISHED'}
 
+class MCPREP_OT_place_json_model_with_gizmo(bpy.types.Operator):
+	bl_idname = "mcprep.place_json_model_with_gizmo"
+	bl_label = "Import Minecraf JSON model and Place"
+	bl_description = "Imports a Minecraft JSON model with location selection"
+	
+	filename_ext = ".json"
+	filepath: bpy.props.StringProperty(subtype='FILE_PATH', options={'SKIP_SAVE'})
+	def invoke(self, context, event):
+		from mathutils import Vector, Quaternion
+
+		self.has_hit = False
+		self.hit_location = Vector((0, 0, 0))
+		self.hit_normal = Vector((0, 0, 1))
+		self.rotation_quat = Quaternion()
```

Comment:
Also thinking it might be a good idea to have a basic dataclass to encapsulate these variables so we can simplify the raycast function arguments, and avoid juggling around 4 variables. 

While I'm at it, I'll look into eliminating these instance variables completely, and just have them as local variables to a function (honestly not a fan of having mutable instance variables in an operator, it just feels weird), though that might be a bit difficult given we have a callback to deal with).